### PR TITLE
Update of nearstars.stc

### PR DESCRIPTION
**Pull request by LukeCEL**
This is an overhaul of the nearstars.stc file. Changes include:

- Updated distances for most stars, mostly Gaia DR2
- Updated parameters, generally from SIMBAD or individual sources
- New orbital elements for many star systems
- Includes spectral type Y, not included in Celestia 1.6.1
- Changed Gliese/GJ naming scheme: 1 to 999 is Gliese; 1000 above is GJ
- Removed star systems that are known to not be within 25 ly

The file still retains the overall structure, as well as part of the heading.

### CODE REVIEW

**Comment by SevenSpheres** [comment]
Noting that this was previously #550

**Comment by LukeCEL** [comment]
> Please fix WISE 0855-0714, it is now 48 solar radii.

Fixed.

> Also, you should probably use rebase to make your three commits into one.

Do I have to?

File: data/nearstars.stc [hunk 1]
```diff
@@ -440,53 +468,56 @@ Barycenter 37279 "Procyon:Elgomaisa:ALF CMi:10 CMi:Gliese 280:ADS 6251"
...
+		Eccentricity     0.40
+		Inclination     42.58
+		AscendingNode   25.26
+		ArgOfPericenter 90.19
+		MeanAnomaly    281.41
 	}
 }
 
-"Procyon B:Elgomaisa B:ALF CMi B:10 CMi B:Gliese 280 B:ADS 6251 B"
+"Procyon B:ALF CMi B:10 CMi B"
```
Comment (375gnu) [inline data/nearstars.stc]: why did you remove Elgomaisa B and Gliese 280 B:ADS 6251 B here?

File: data/nearstars.stc [hunk 2]
```diff
@@ -595,743 +646,760 @@ Barycenter "EPS Ind B:CI Ind:Gliese 845 B"
...
 "2MASS 0939-2448 A:2MASS J09393548-2448279 A"
 {
-	Texture "exo-class4night.*"
-	RA 144.897823
-	Dec -24.80775
-	Distance 17.414
+	RA 144.897865
+	Dec -24.807761
+	Distance 17.414 # 187.30 +/- 4.60 mas
+	Texture "exo-class5night.*"
```
Comment (375gnu) [inline data/nearstars.stc]: planet texture assigned to a star?

File: data/nearstars.stc [hunk 3]
```diff
@@ -1340,244 +1408,244 @@ Barycenter "GJ 1116"  # orbit of A & B unknown
...
 "2MASS 0939-2448 B:2MASS J09393548-2448279 B"
 {
-	Texture "exo-class4night.*"
-	RA 144.897843
-	Dec -24.80775
-	Distance 17.414
+	RA 144.897885
+	Dec -24.807761
+	Distance 17.414 # 187.30 +/- 4.60 mas
+	Texture "exo-class5night.*"
```
Comment (375gnu) [inline data/nearstars.stc]: as above

File: data/nearstars.stc [hunk 4]
```diff
@@ -1340,244 +1408,244 @@ Barycenter "GJ 1116"  # orbit of A & B unknown
...
+# radius and temperature from Line et al. (2017), ApJ 848 (2), 83
+# "Uniform Atmospheric Retrieval Analysis of Ultracool Dwarfs. II. Properties 
+# of 11 T dwarfs"
+# http://iopscience.iop.org/0004-637X/848/2/83/
+"2MASS 1114-2618:2MASS J11145133-2618235"
+{
+	RA 168.713904
+	Dec -26.306544
+	Distance 18.45 # 176.8 +/- 7.0 mas
+	Texture "exo-class5night.*"
```
Comment (375gnu) [inline data/nearstars.stc]: and here

File: data/nearstars.stc [hunk 5]
```diff
@@ -1340,244 +1408,244 @@ Barycenter "GJ 1116"  # orbit of A & B unknown
...
+# http://iopscience.iop.org/article/10.1088/0004-637X/762/2/119/meta
+# rotation period, radius from Burningham et al. (2016), MNRAS 463, 2
+# "A LOFAR mini-survey for low-frequency radio emission from the nearest brown dwarfs"
+# https://academic.oup.com/mnras/article/463/2/2202/2892466
+"WISE 1741+2553:WISE J174124.25+255319.6"
+{
+	RA 265.352586
+	Dec 25.892878
+	Distance 18.53 # 176 +/- 25 mas
+	Texture "exo-class5night.*"
```
Comment (375gnu) [inline data/nearstars.stc]: and here

File: data/nearstars.stc [hunk 6]
```diff
@@ -1340,244 +1408,244 @@ Barycenter "GJ 1116"  # orbit of A & B unknown
...
 "2MASS 0415-0935:2MASS J04151954-0935066"
 {
-	Texture "exo-class4night.*"
-	RA 63.83125
-	Dec -9.585
-	Distance 18.709
+	RA 63.8381022
+	Dec -9.5835266
+	Distance 18.62 # 175.2 +/- 1.7 mas
+	Texture "exo-class5night.*"
```
Comment (375gnu) [inline data/nearstars.stc]: here again

File: data/nearstars.stc [hunk 7]
```diff
@@ -2450,294 +2692,291 @@ Barycenter "GJ 3522 A:Giclas 41-14 A"
...
+# low-quality parallax from Theissen, C. A. (2018), ApJ 862 (2), 173
+# "Parallaxes of Cool Objects with WISE: Filling in for Gaia"
+# http://iopscience.iop.org/article/10.3847/1538-4357/aaccfa/meta
+"WISE 0254+0223:WISE J025409.51+022358.6"
 {
-	Radius 305000 # approx. for class
+	RA 43.5328583
+	Dec 2.3989861
+	Distance 23.5 # 139 +/- 40 mas
+	Texture "exo-class5night.*"
```
Comment (375gnu) [inline data/nearstars.stc]: again here

File: data/nearstars.stc [hunk 8]
```diff
@@ -2986,279 +3267,176 @@ Barycenter 5336 "MU Cas:30 Cas:Gliese 53"
...
+Modify 113368 # Fomalhaut 
 {
 	UniformRotation
 	{
-	    Inclination    81.8    # rotation pole matches debris disk
-	    AscendingNode  176.6   # Bouquin et al. (2009) A&A 498, L41-44
-# (http://www.aanda.org/index.php?option=article&access=doi&doi=10.1051/0004-6361/200911854)
+		Inclination    81.8    # rotation pole matches debris disk
+		AscendingNode  176.6   # Bouquin et al. (2009) A&A 498, L41-44
+		# (http://www.aanda.org/index.php?option=article&access=doi&doi=10.1051/0004-6361/200911854)
```
Comment (375gnu) [inline data/nearstars.stc]: please undo whitespace change

File: data/nearstars.stc [hunk 9]
```diff
@@ -2986,279 +3267,176 @@ Barycenter 5336 "MU Cas:30 Cas:Gliese 53"
...
 		AscendingNode   192.13
-		ArgOfPericenter 189.69
-		MeanAnomaly     29.62
+		ArgOfPericenter 189.70
+		MeanAnomaly      29.62
 	}
 }
 
-"Alula Australis Aa:XI UMa Aa:53 UMa Aa:Gliese 423 Aa"
+"Alula Australis Aa:XI UMa Aa:53 UMa Aa:Gliese 423 Aa"
```
Comment (375gnu) [inline data/nearstars.stc]: please remove trailing whitespces. here and below

**Comment by LukeCEL** [comment]
> please undo whitespace change here

> please remove trailing whitespces. here and below

> why did you remove Elgomaisa B and Gliese 280 B:ADS 6251 B here?

These have all been fixed.

> planet texture assigned to a star?

Yep, this was actually Grant Hutchinson's idea (see [this](https://github.com/CelestiaProject/Celestia/commit/3f8d95466ee3e2271ab75c6c0ab18931eb89fc58#diff-0e8eba7077d701f35537deb7f190ae0a) edit). But for now I've removed all texture specifications (except for two times where `"browndwarf.*"` is used). So I guess we can argue about which textures to use over at PR #477.

**Comment by SevenSpheres** [comment]
For some reason I can't make a review now...
- There are still two brown dwarfs with Hot Jupiter nightside textures
- Please remove the "browndwarf.*" texture also
- I still recommend using rebase

**Comment by LukeCEL** [comment]
> There are still two brown dwarfs with Hot Jupiter nightside textures
> Please remove the "browndwarf.*" texture also

Done.

> I still recommend using rebase

Okay but why. I don't really see the use for it.

**Comment by 375gnu** [comment]
> Okay but why. I don't really see the use for it.

Because we want to have clear history. So please squash your commits into a single one.
```
git rebase -i HEAD~N # N is number of commits in your PR
# replace all but the first pick with f
git push -f
```
Please also expand the commit message to include description from above:
```
short description

much longer description
```

You can do this during rebase if you replace the 1st `pick` with `edit`.

Thaks.

**Comment by LukeCEL** [comment]
> Because we want to have clear history. 

Understood.

> So please squash your commits into a single one.

Apologies, I am currently working out how to do this, since my fork started to fall behind in terms of the changes.

**Comment by LukeCEL** [comment]
I think I did it. Can you both check?

**Comment by SevenSpheres** [comment]
> I think I did it. Can you both check?

Looks good.

**Comment by SevenSpheres** [comment]
This has already been merged, but I found another problem: "G 141-36" should be "Giclas 141-36".
